File: MakeMeLook/чистая версия/mml-saas-backend/internal/dto/analytics.go
package dto

// ── Summary (KPI) ──────────────────────────────────────────────────────────

type AnalyticsKPI struct {
	Value int64   `json:"value"`
	Trend float64 `json:"trend"` // % change vs previous period
}

type AnalyticsSummaryResponse struct {
	Views      AnalyticsKPI `json:"views"`
	Opens      AnalyticsKPI `json:"opens"`
	TryOns     AnalyticsKPI `json:"tryons"`
	Leads      AnalyticsKPI `json:"leads"`
	Conversion AnalyticsKPI `json:"conversion"`
	CartItems  AnalyticsKPI `json:"cart_items"`
	Revenue    AnalyticsKPI `json:"revenue"`
	AOV        AnalyticsKPI `json:"aov"`
}

// ── Funnel ─────────────────────────────────────────────────────────────────

type FunnelStageResponse struct {
	Key   string  `json:"key"`
	Count int64   `json:"count"`
	Pct   float64 `json:"pct"` // % relative to first stage
}

type AnalyticsFunnelResponse struct {
	Stages []FunnelStageResponse `json:"stages"`
}

// ── Trends ─────────────────────────────────────────────────────────────────

type TrendPointResponse struct {
	Date  string `json:"date"`
	Value int64  `json:"value"`
}

type AnalyticsTrendsResponse struct {
	Metric string               `json:"metric"`
	Points []TrendPointResponse `json:"points"`
}

// ── Top Products ───────────────────────────────────────────────────────────

type TopProductResponse struct {
	ProductID int    `json:"product_id"`
	Name      string `json:"name"`
	PhotoURL  string `json:"photo_url"`
	TryOns    int64  `json:"tryons"`
	Favorites int64  `json:"favorites"`
	Cart      int64  `json:"cart"`
}

type AnalyticsTopProductsResponse struct {
	SortBy   string               `json:"sort_by"`
	Products []TopProductResponse `json:"products"`
}

// ── Audience ───────────────────────────────────────────────────────────────

type DistributionItem struct {
	Name  string `json:"name"`
	Value int64  `json:"value"`
}

type AnalyticsAudienceResponse struct {
	Gender      []DistributionItem `json:"gender"`
	Sizes       []DistributionItem `json:"sizes"`
	FigureTypes []DistributionItem `json:"figure_types"`
	Devices     []DistributionItem `json:"devices"`
}

File: MakeMeLook/docker-wedding/mml-saas-frontend/src/shared/api/analytics.ts
import { apiClient } from './client';

// ─── Types ──────────────────────────────────────────────────────────────────

export interface AnalyticsKPI {
  value: number;
  trend: number;
}

export interface AnalyticsSummary {
  views: AnalyticsKPI;
  opens: AnalyticsKPI;
  tryons: AnalyticsKPI;
  leads: AnalyticsKPI;
  conversion: AnalyticsKPI;
  cart_items: AnalyticsKPI;
  revenue: AnalyticsKPI;
  aov: AnalyticsKPI;
}

export interface FunnelStage {
  key: string;
  count: number;
  pct: number;
}

export interface AnalyticsFunnel {
  stages: FunnelStage[];
}

export interface TrendPoint {
  date: string;
  value: number;
}

export interface AnalyticsTrends {
  metric: string;
  points: TrendPoint[];
}

export interface TopProduct {
  product_id: number;
  name: string;
  photo_url: string;
  tryons: number;
  favorites: number;
  cart: number;
}

export interface AnalyticsTopProducts {
  sort_by: string;
  products: TopProduct[];
}

export interface DistributionItem {
  name: string;
  value: number;
}

export interface AnalyticsAudience {
  gender: DistributionItem[];
  sizes: DistributionItem[];
  figure_types: DistributionItem[];
  devices: DistributionItem[];
}

// ─── API ────────────────────────────────────────────────────────────────────

export const analyticsApi = {
  getSummary: async (
    projectId: number,
    period: string,
  ): Promise<AnalyticsSummary> => {
    const response = await apiClient.get<AnalyticsSummary>(
      `/api/v1/projects/${projectId}/analytics/summary`,
      { params: { period } },
    );
    return response.data;
  },

  getFunnel: async (
    projectId: number,
    period: string,
  ): Promise<AnalyticsFunnel> => {
    const response = await apiClient.get<AnalyticsFunnel>(
      `/api/v1/projects/${projectId}/analytics/funnel`,
      { params: { period } },
    );
    return response.data;
  },

  getTrends: async (
    projectId: number,
    period: string,
    metric: string,
  ): Promise<AnalyticsTrends> => {
    const response = await apiClient.get<AnalyticsTrends>(
      `/api/v1/projects/${projectId}/analytics/trends`,
      { params: { period, metric } },
    );
    return response.data;
  },

  getTopProducts: async (
    projectId: number,
    period: string,
    sortBy: string,
  ): Promise<AnalyticsTopProducts> => {
    const response = await apiClient.get<AnalyticsTopProducts>(
      `/api/v1/projects/${projectId}/analytics/top-products`,
      { params: { period, sort_by: sortBy } },
    );
    return response.data;
  },

  getAudience: async (
    projectId: number,
    period: string,
  ): Promise<AnalyticsAudience> => {
    const response = await apiClient.get<AnalyticsAudience>(
      `/api/v1/projects/${projectId}/analytics/audience`,
      { params: { period } },
    );
    return response.data;
  },
};
